core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/store/dot-experiments-configuration-store.ts

```diff
@@ -415,6 +424,54 @@ export class DotExperimentsConfigurationStore extends ComponentStore<DotExperime
         }
     );
 
+    readonly setDescription = this.effect(
+        (
+            description$: Observable<{
+                experiment: DotExperiment;
+                data: Pick<DotExperiment, 'description'>;
+            }>
+        ) => {
+            return description$.pipe(
+                tap(() =>
+                    this.setSidebarStatus({
+                        status: ComponentStatus.SAVING,
+                        experimentStep: ExperimentSteps.EXPERIMENT_DESCRIPTION
+                    })
+                ),
+                switchMap(({ experiment, data }) =>
+                    this.dotExperimentsService.setDescription(experiment.id, data.description).pipe(
+                        tapResponse(
+                            (response) => {
+                                this.messageService.add({
+                                    severity: 'info',
+                                    summary: this.dotMessageService.get(
+                                        'experiments.configure.description.edit.confirm-title'
+                                    ),
+                                    detail: this.dotMessageService.get(
+                                        'experiments.configure.description.edit.confirm-message',
+                                        experiment.name
+                                    )
+                                });
+
+                                this.setExperiment(response);
+                                this.setSidebarStatus({
+                                    status: ComponentStatus.IDLE
+                                });
+                            },
+
+                            (error: HttpErrorResponse) => {
+                                this.setSidebarStatus({
+                                    status: ComponentStatus.IDLE
+                                });
+                                this.dotHttpErrorManagerService.handle(error);
+                            }
+                        )
+                    )
+                )
```

Comment:
Since we use `switchMap`, I think there's no need to neste a pipe.

```suggestion
               switchMap(({ experiment, data }) =>
this.dotExperimentsService.setDescription(experiment.id, data.description),
                tapResponse(
                    (response) => {
                        this.messageService.add({
                            severity: 'info',
                            summary: this.dotMessageService.get(
                                'experiments.configure.description.edit.confirm-title'
                            ),
                            detail: this.dotMessageService.get(
                                'experiments.configure.description.edit.confirm-message',
                                experiment.name
                            )
                        });

                        this.setExperiment(response);
                    },

                    (error: HttpErrorResponse) => this.dotHttpErrorManagerService.handle(error),
                    
                    // Finalize
                    () => {
                        this.setSidebarStatus({
                            status: ComponentStatus.IDLE
                        });
                    }
                )
```

Reply:
the nested pipe needed.